nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/diagnostics/bootstrap/tasks/OperatingSystemDiagnosticTask.java

```diff
@@ -69,11 +74,26 @@ public DiagnosticsDumpElement captureDump(final boolean verbose) {
             }
 
             attributes.forEach((key, value) -> details.add(key + " : " + value));
+            details.addAll(getPhysicalCPUCores());
+            details.addAll(getTotalPhysicalRam());
+            details.addAll(getDiskLayout());
         } catch (final Exception e) {
             logger.error("Failed to obtain Operating System details", e);
             return new StandardDiagnosticsDumpElement("Operating System / Hardware", Collections.singletonList("Failed to obtain Operating System details"));
         }
 
         return new StandardDiagnosticsDumpElement("Operating System / Hardware", details);
     }
+
+    private Collection<String> getPhysicalCPUCores() {
+        return ShellCommandExecutor.execute(new GetPhysicalCpuCoresCommand());
```

Comment:
After refactoring, this should be changed to create the command instance and then call `execute()` instead of calling `ShellCommandExecutor.execute()`.